src/borg/locking.py

```diff
@@ -21,6 +23,29 @@ def get_id():
     return _hostname, _pid, thread_id
 
 
+def check_lock_data(host, pid, thread):
+    """Check if the host, pid, thread combination corresponds to a dead process on our local node or not."""
+    if host != _hostname:
+        return False
+
+    try:
+        # This may not work in Windows.
+        # This does not kill anything, 0 means "see if we can send a signal to this process or not".
+        # Possible errors: No such process (== stale lock) or permission denied (not a stale lock)
+        # If the exception is not raised that means such a pid is valid and we can send a signal to it (== not a stale lock too).
+        os.kill(pid, 0)
+        return False
+    except OSError as err:
+        if err.errno != errno.ESRCH:
+            return False
+        pass
+
+    # We should be ok not checking thread no matter what it is, right?
+    # Possible implications on Windows?
```

Comment:
So, should I instead just ensure that thread is always 0, if it's not 0, something way too new is going on and we return False at the onstart?